### PR TITLE
Update versioning of StoneWork images

### PR DESCRIPTION
**Pull request by mhalaj1**
Signed-off-by: mhalaj1 <matus.halaj@pantheon.tech>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -124,10 +139,10 @@ cleanup-images:
 vm-image: images
 	@echo "=> building stonework VM image"
 ifdef CNFS_SPEC
-	VERSION=${VPP_VERSION} \
+	VERSION=${VERSION} \
 	./scripts/vm/create-vm-image.sh --cnfs-spec ${CNFS_SPEC}
 else
-	VERSION=${VPP_VERSION} \
+	VERSION=${VERSION} \
```
Comment (mhalaj1) [inline Makefile:145]: I am not sure if I updated the lines in `vm-image` target correctly

File: docker/dev.Dockerfile [hunk 2]
```diff
@@ -14,7 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ARG VPP_IMAGE=vpp:21.06
+ARG VPP_IMAGE=ghcr.io/pantheontech/vpp
```
Comment (ondrej-fabry) [inline docker/dev.Dockerfile:17]: This image does not exist though.

File: Makefile [hunk 3]
```diff
@@ -21,6 +21,15 @@ DATE    ?= $(shell git log -1 --format="%ct" | xargs -I{} date -d @{} +'%Y-%m-%d
 CNINFRA := go.ligato.io/cn-infra/v2/agent
 LDFLAGS = -X $(CNINFRA).BuildVersion=$(VERSION) -X $(CNINFRA).CommitHash=$(COMMIT) -X $(CNINFRA).BuildDate=$(DATE)
 
+RELEASE_TAG ?= $(shell git describe --always --tags --dirty --exact-match 2>/dev/null)
```
Comment (ondrej-fabry) [inline Makefile:24]: The `--dirty` should be omitted in this case to avoid naming the release with `-dirty` suffix.

Comment (mhalaj1) [reply Makefile:24]: Added the check to throw error if repo is dirty.

**Comment by mhalaj1** [comment]
This PR also fixes #4 (the documentation on building `dev-image` is corrected).

File: README.md [hunk 4]
```diff
@@ -237,7 +237,7 @@ volumes:
 services:
   stonework:
     container_name: stonework
-    image: "ghcr.io/pantheontech/stonework"
+    image: "ghcr.io/pantheontech/stonework:21.06.0"
```
Comment (ondrej-fabry) [inline README.md:240]: The tag should actually be `21.06` - rolling release which includes patch releases as well.

File: Makefile [hunk 5]
```diff
@@ -21,17 +21,29 @@ DATE    ?= $(shell git log -1 --format="%ct" | xargs -I{} date -d @{} +'%Y-%m-%d
...
+ifneq ($(RELEASE_TAG_CHECKED),)
+RELEASE_VERSION_FULL = $(shell echo $(RELEASE_TAG_CHECKED) | cut -c 2-)
+RELEASE_VERSION_MAJOR_MINOR = $(shell echo $(RELEASE_VERSION_FULL) | cut -d '.' -f 1-2)
+endif
+
 ifeq ($(VPP_VERSION),)
 VPP_VERSION="21.06"
 endif
+ifeq ($(DEV_VERSION),) # for tagging in-development images
+DEV_VERSION="21.06.0"
```
Comment (ondrej-fabry) [inline Makefile:37]: This should be `21.06`. The `21.06.0` tag is a unique tag which is built and pushed only once on the actual git tag `v21.06.0`. Unlike the `21.06` tag, which is a rolling tag that gets updated to the latest patch release of `v21.06.XX` version.

File: Makefile [hunk 6]
```diff
@@ -58,64 +70,72 @@ install-proto-rootgen:
 # -------------------------------
 
 vpp-image:
-	@echo "=> building VPP image -- version=${VPP_VERSION}"
-	IMAGE_TAG=${STONEWORK_VPP_IMAGE} \
-	VERSION=${VPP_VERSION} \
+	@echo "=> building VPP image"
+	VPP_VERSION=${VPP_VERSION} \
+	IMAGE_TAG="${STONEWORK_VPP_IMAGE}:${DEV_VERSION}" \
```
Comment (ondrej-fabry) [inline Makefile:75]: This once should be tagged with VPP version, so it follows the same versioning as vpp-base does. No point in tagging vpp image with same tag as stonework.

**Review by ondrej-fabry — APPROVED** [review]

**Pull request merged**